source/grid/grid_generator.cc

```diff
@@ -337,18 +337,19 @@ namespace GridGenerator
   template <int dim, int spacedim>
   void
   hyper_rectangle (Triangulation<dim,spacedim> &tria,
-                   const Point<spacedim>   &p_1,
-                   const Point<spacedim>   &p_2,
+                   const Point<dim>   &p_1,
+                   const Point<dim>   &p_2,
                    const bool          colorize)
   {
-    // First, normalize input such that
-    // p1 is lower in all coordinate directions.
-    Point<spacedim> p1(p_1);
-    Point<spacedim> p2(p_2);
-
-    for (unsigned int i=0; i<spacedim; ++i)
-      if (p1(i) > p2(i))
-        std::swap (p1(i), p2(i));
+    // First, extend dimensions from dim to spacedim and
+    // normalize such that p1 is lower in all coordinate
+    // directions. Additional entries will be 0.
+    Point<spacedim> p1, p2;
+    for (unsigned int i=0; i<dim; ++i)
+      {
+        p1(i) = std::min(p_1(i), p_2(i));
+        p2(i) = std::max(p_1(i), p_2(i));
```

Comment:
So this means that if you give `(1,0)` and `(0,1)` as corner points, that you get the unit square? But then I see that that's already how it was before.